### PR TITLE
WIP global: initial module

### LINKED ISSUE #1: global: initial implementation
Create a module similar to Invenio-OpenAIRE that can import http://licenses.opendefinition.org/licenses/groups/all.json

Required:
- [ ] JSON schema for licenses.
- [ ] Elasticsearch mapping.
- [ ] CLI command ``<cmd> opendefinition loadlicenses``
- [ ] Celery tasks: 1) Download JSON and send tasks to create records 2) Create records + PIDs
- [ ] JSONResolver so that we can resolve ``"license": {"$ref": "http://dx.zenodo.org/licenses/<pid>"}`` without making HTTP requests.

### PR DESCRIPTION
**Pull request by SamiHiltunen**
* Initial release of Invenio-OpenDefinition. (closes #1)

Signed-off-by: Sami Hiltunen <sami.mikael.hiltunen@cern.ch>

### CODE REVIEW

File: invenio_opendefinition/config.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Invenio module integrating Invenio repositories and OpenDefinition."""
+
+OPENDEFINITION_LICENSES_URL = 'http://licenses.opendefinition.org/licenses'
```
Comment (lnielsen) [inline invenio_opendefinition/config.py]: Style:
```python
OPENDEFINITION_LICENSES_URL = \
    'http://licenses.opendefinition.org/licenses/groups/all.json'
```

File: invenio_opendefinition/ext.py [hunk 2]
```diff
@@ -22,13 +22,13 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module providing integration between Invenio repositories and OpenDefinition."""
+"""Invenio module integrating Invenio repositories and OpenDefinition."""
 
 from __future__ import absolute_import, print_function
 
 from flask_babelex import gettext as _
```
Comment (lnielsen) [inline invenio_opendefinition/ext.py]: I think you can completely remove this one.

File: invenio_opendefinition/tasks.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+from .minters import license_minter
+from .resolvers import license_resolver
+from .validators import license_validator
+
+
+@shared_task(ignore_result=True)
+def harvest_licenses():
+    """Harvest OpenDefinition licenses."""
+    response = requests.get(current_app.config['OPENDEFINITION_LICENSES_URL'])
+    licenses = json.loads(response.text)
```
Comment (lnielsen) [inline invenio_opendefinition/tasks.py]: ``licenses = respose.json()``

File: invenio_opendefinition/tasks.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    for license in licenses:
+        license_validator.validate(license)
+        create_or_update_license_record.delay(licenses[license])
+
+
+@shared_task(ignore_result=True)
+def create_or_update_license_record(license):
+    """Register a license."""
+    try:
+        pid, record = license_resolver.resolve(license['id'])
```
Comment (lnielsen) [inline invenio_opendefinition/tasks.py]: are you injecting ``$schema`` somewhere?

File: setup.py [hunk 5]
```diff
@@ -48,20 +49,39 @@
...
-    'Babel>=1.3',
+    'babel>=1.3',
     'pytest-runner>=2.6.2',
 ]
 
 install_requires = [
-    'Flask-BabelEx>=0.9.2',
+    'click>=6.4',
+    'flask-babelex>=0.9.2',
+    'invenio-celery>=1.0.0a4',
```
Comment (lnielsen) [inline setup.py]: it might be enough to just depend on flask-celeryext

**Comment by lnielsen** [comment]
Test error:
-  ``jsonschema.exceptions.RefResolutionError: No JSON object could be decoded`` Just set JSONSCHEMAS_HOST`` to zenodo.org or use localhost instead.

**Comment by SamiHiltunen** [comment]
@lnielsen Could you go over this and give comments or merge as needed.

**Pull request merged**